src/acl/Acl.cc

```diff
@@ -268,22 +305,11 @@ Acl::Node::ParseAclLine(ConfigParser &parser, Node ** head)
         new_acl = 0;
     }
 
-    /*
-     * Here we set AclMatchedName in case we need to use it in a
-     * warning message in Acl::SplayInserter::Merge().
-     */
-    AclMatchedName = A->name;   /* ugly */
```

Comment:
Removal of this "ugly" hack (and the code that uses it) is the focus of this PR, but I expect this PR changes to have long-term positive effects on configuration parsing in general. BTW, the work on this PR also triggered discovery of problems fixed in recent commit 0898d0f; we had even tried to fix them in this PR (before we realized how deep that well goes).